File: Luna.Web/_wwwsrc/js/Addons/Models/Character.ts
import {CharacterType} from "./CharacterType";
import {Race} from "./Race";
import {CustomSection} from "./CustomSection";

export class Character {
    public id: number;
    public name: string;
    public description: string;
    public type: CharacterType;
    public race: Race;
    public customSections: CustomSection[];
    
    constructor(data?) {
        if (data) {
            this.id = data.id;
            this.name = data.name;
            this.description = data.description;
            this.type = new CharacterType(data.type);
            this.race = new Race(data.race);
            this.customSections = data.customSections ? data.customSections.map(_ => new CustomSection(_)) : [];
        } else {
            this.customSections = [];
        }
    }
}

File: Luna.Commons/Models/Character.cs
using System.Collections.Generic;

namespace Luna.Commons.Models
{
    public class Character : ModelBase
    {
        public virtual int TypeId { get; set; }
        public virtual int RaceId { get; set; }
        
        public virtual CharacterType Type { get; set; }
        public virtual Race Race { get; set; }
        public virtual IList<CustomSection> CustomSections { get; set; }
    }
}